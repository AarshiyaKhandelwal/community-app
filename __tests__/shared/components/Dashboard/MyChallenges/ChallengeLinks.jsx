import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import ChallengeLinks from 'components/Dashboard/MyChallenges/ChallengeLinks';

const mockData1 = {
  viewMode: 'list',
  challenge: {
    track: 'DEVELOP',
    subTrack: 'CODE',
  },
};

const mockData2 = {
  viewMode: 'list',
  challenge: {
    track: 'DEVELOP',
    subTrack: 'MARATHON_MATCH',
    rounds: [{}],
    numRegistrants: [0],
  },
};

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  renderer.render((
    <ChallengeLinks {...mockData1} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <ChallengeLinks {...mockData2} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
