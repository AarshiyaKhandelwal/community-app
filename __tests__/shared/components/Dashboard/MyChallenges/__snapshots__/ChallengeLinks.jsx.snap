// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__challenge-links___1fJfh src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__list-view___2mV4T"
>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__registrants___3kbDS"
    href="https://www.topcoder-dev.com/challenge-details/undefined/?type=develop#viewRegistrant"
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__registrants-icon___tm0l6"
    />
    <p />
  </a>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__submissions___3TDly"
    href="https://www.topcoder-dev.com/challenge-details/undefined/?type=develop#submissions"
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__submissions-icon___3MZ-_"
    />
    <p />
  </a>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__forum___TQDgX"
    href="https://apps.topcoder-dev.com/forums/?module=Category&categoryID=undefined"
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__forum-icon___b72Li"
    />
    <p>
      Posts
    </p>
  </a>
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__challenge-links___1fJfh src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__list-view___2mV4T"
>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__registrants___3kbDS"
    href="https://community.topcoder-dev.com/longcontest/?module=ViewRegistrants&rd=undefined"
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__registrants-icon___tm0l6"
    />
    <p />
  </a>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__submissions___3TDly"
    href=""
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__submissions-icon___3MZ-_"
    />
    <p />
  </a>
  <a
    className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__forum___TQDgX"
    href="https://apps.topcoder-dev.com/forums/?module=ThreadList&forumID=undefined"
  >
    <div
      className="src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__icon___3IZGz src-shared-components-Dashboard-MyChallenges-ChallengeLinks-___style__forum-icon___b72Li"
    />
    <p>
      Posts
    </p>
  </a>
</div>
`;
